core/app/models/spree/variant/scopes.rb

```diff
@@ -10,6 +10,14 @@ def self.prepended(base)
             order(Arel.sql("COALESCE((SELECT COUNT(*) FROM  #{Spree::LineItem.quoted_table_name} GROUP BY #{Spree::LineItem.quoted_table_name}.variant_id HAVING #{Spree::LineItem.quoted_table_name}.variant_id = #{Spree::Variant.quoted_table_name}.id), 0) DESC"))
           }
 
+          scope :by_stock_location, ->(stock_location_id) {
+            if stock_location_id
+              joins(:stock_locations).where(spree_stock_locations: { id: stock_location_id })
+            else
+              all
```

Comment:
I believe with scopes, if `nil` is returned it essentially acts as an `all` anyways.

Reply:
@BenMorganIO do you see any issue using `all` here?

Reply:
No, I just thought to point out the `else` section might be able to be deleted entirely.

Reply:
Ah, cause it will fallback to `nil` and will act as `.all`. Up to @elia (currently assigned to this PR) if he wants to change it.

Reply:
@BenMorganIO @kennyadsl good call, I updated the code to avoid the check altogether and added a spec for `Variant.by_stock_location`.